jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
Hmm I see your earlier message mentioning that it only fails at runtime because `__getnewargs__` needs an `int` argument so I see the reasoning for actually wanting to call the method to see if it actually works. That's a bummer.

Is there any way to encode/know exactly what that `int` arg should be so that we don't have to guess?

Also, it would be good for the test cases to include an actual Enum test case since that's the problem we're trying to fix. Verifying that the actual enum value makes it through the round-trip is what we should verify.